File: algorithms/binary-search/Python/src/binary_search.py
import math


def binary_search(arr: list, target: int) -> int:
    if len(arr) == 0:
        return -1

    left = 0
    right = len(arr) - 1

    while (left <= right):
        MID = math.floor((left + right) / 2)
        GUESS = arr[MID]

        if GUESS == target:
            return MID
        if (GUESS > target):
            right = MID - 1
        else:
            left = MID + 1

    return -1

File: Java/src/algorithms/binary-search/BinarySearch.java
import java.util.ArrayList;

public class BinarySearch {
  public static void main() {
  }

  public static int binarySearch(ArrayList<Integer> arr, int target) {
    if (arr.size() == 0) {
      return -1;
    }

    int right = 0;
    int left = arr.size() - 1;

    while (left <= right) {
      final int mid = (int) Math.floor((right + left) / 2);
      final int guess = arr.get(mid);
      if (guess == target) {
        return mid;
      }
      if (guess > target) {
        right = mid - 1;
      } else {
        left = mid + 1;
      }
    }
    return -1;
  }
}
